packages/babel-parser/src/parser/expression.ts

```diff
@@ -2746,15 +2749,17 @@ export default abstract class ExpressionParser extends LValParser {
     name: string,
   ): N.Identifier {
     node.name = name;
-    node.loc.identifierName = name;
+    if (this.optionFlags & OptionFlags.Locations) {
+      node.loc.identifierName = name;
+    }
```

Comment:
This change seems to be a breaking one. Can we defer it to Babel 8?

Reply:
The no-locations mode is opt-in, the default behavior is still to include an `identifierName`.